src/mixins/socket.js

```diff
@@ -150,7 +150,16 @@ export default {
                         }
                     };
                 });
-                this.monitorList = data;
+
+                if (data.op === OPERATIONS.ADD) {
+                    this.monitorList = this.updateMonitorList(data.list);
+                } else if (data.op === OPERATIONS.UPDATE) {
+                    this.monitorList = this.updateMonitorList(data.list);
+                } else if (data.op === OPERATIONS.DELETE) {
+                    this.monitorList = this.deleteMonitorList(data.monitorID);
```

Comment:
this is currently a no-op, right?

Reply:
@CommanderStorm 

i don't understand no-op.

do you means not in used?

we are using this.

let me explain why op is required.

current scenario what will happened in case of add/update/delete
lets assume 400 URLs

1) page-load: fetching **400urls** & render
2) add-monitor: time for add into database + fetching **401urls** from database + building monitor.json + update list this.monitorList = data.list
3) update-monitor: time for update into database + fetching **401urls** from database + building monitor.json + update list this.monitorList = data.list
4) delete-monitor: time for delete into database + fetching **400urls** from database + building monitor.json + update list this.monitorList = data.list

after adding op parameter
1) page-load:  fetching 400urls from database + building monitor.json + update list this.monitorList = data.list
2) add-monitor: time for add into database + fetching **1 url** from database + building monitor.json + update list this.monitorList = data.list
3) update-monitor: time for update into database + fetching **1 url** from database + building monitor.json + update list this.monitorList = data.list
4) delete-monitor: time for delete into database + fetching **1 url** from database + building monitor.json + update list this.monitorList = data.list

it will make huge difference

page load time will be same but add/update/delete time will be < 1 sec instead of 11 sec

Reply:
> i don't understand no-op.

I mean https://wikipedia.org/wiki/NOP_(code)

Honestly, `DELETE` being mingled in next to `LIST`, `UPDATE` and `ADD` is quite messy and has caused by some confusion in the review.
As noted by you, they should be different. I don't think that differentiating via `op` is good though. Adding this, means that the spagetti-code pile gets larger

=> It would be better from a reasoning about code perspective to remove the code and make this into different endpoints.
This also allows furhter (operation specific) optimisations for creation, update, list and deletion.
I like [googles guidelines](https://cloud.google.com/apis/design/standard_methods) for this.

Reply:
@CommanderStorm 

sorry for being late reply, i was a little busy.

ultimately your approach to revert these changes as per google guidelines?
am I correct?
if yes then might be can not reduce time while insert/update/delete. because anyhow we can not reduce unnecessary processing.

correct me if my understanding is wrong.

Reply:
Not revert, but rework.
Splitting this into separate socketio-endpoints seems like a good call.

Currently everything is muddled into the same list endpoint.
Switching functionality off with `op` seems like somewhat of a bodge..

Are we taking past each other?

Reply:
@CommanderStorm 
i understand your point.
Splitting this into separate socketio-endpoints seems like a good call.

let me start work on that. if get time today I will update pr. its last changes other changes already done

Reply:
@CommanderStorm 
i updated as per your instructions. can you confirm?